x-pack/plugins/observability_solution/observability_onboarding/public/assets/auto_detect.sh

```diff
@@ -0,0 +1,177 @@
+}
+
+detect_known_integrations() {
+  while IFS= read -r log_file_path; do
+    local integration=""
+
+    if [[ $log_file_path =~ ^(/private)?/var/log/nginx ]]; then
+      integration="nginx"
+    fi
+    if [[ $log_file_path =~ ^(/private)?/var/log/(apache2|httpd) ]]; then
+      integration="apache"
+    fi
+    if [[ $log_file_path =~ ^(/private)?/var/lib/docker/containers ]]; then
+      integration="docker"
+    fi
+    if [[ $log_file_path =~ ^(/private)?/var/log/(syslog|auth|system|messages|secure) ]]; then
+      integration="system"
+    fi
+
+    if [ -n "$integration" ]; then
+      known_integrations_list_string+="$integration\n"
+    else
+      unknown_log_file_path_list_string+="$log_file_path\n"
+    fi
+  done <<< "$(echo -e $log_file_list_string)"
+
+  known_integrations_list_string=$(echo -e "$known_integrations_list_string" | sort -u)
+}
+
+known_integration_title() {
+  local integration=$1
+  case $integration in
+    "nginx")
+      echo "Nginx Logs"
+      ;;
+    "apache")
+      echo "Apache Logs"
+      ;;
+    "docker")
+      echo "Docker Container Logs"
+      ;;
+    "system")
+      echo "System Logs"
+      ;;
+    *)
+      echo "Unknown"
+      ;;
+  esac
+}
+
+build_unknown_log_file_patterns() {
+  while IFS= read -r log_file_path; do
+    unknown_log_file_pattern_list_string+="$(dirname "$log_file_path")/*.log\n"
+  done <<< "$(echo -e $unknown_log_file_path_list_string)"
+
+  unknown_log_file_pattern_list_string=$(echo -e "$unknown_log_file_pattern_list_string" | sort -u)
+}
+
+function select_list() {
+  local known_integrations_options=()
+  local unknown_logs_options=()
+
+  while IFS= read -r line; do
+    if [[ -z "$line" ]]; then
+      continue
+    fi
+    known_integrations_options+=("$line")
+  done <<< "$known_integrations_list_string"
+
+  while IFS= read -r line; do
+    if [[ -z "$line" ]]; then
+      continue
+    fi
+    unknown_logs_options+=("$line")
+  done <<< "$unknown_log_file_pattern_list_string"
+
+  local options=("${known_integrations_options[@]}" "${unknown_logs_options[@]}")
+
+  for i in "${!options[@]}"; do
+    if [[ "$i" -lt "${#known_integrations_options[@]}" ]]; then
+      printf "\e[32m%s)\e[0m %s\n" "${i}" "$(known_integration_title "${options[$i]}")"
+    else
+      printf "\e[32m%s)\e[0m %s\n" "${i}" "${options[$i]}"
+    fi
+  done
+
+  echo -e "\n"
+  read -p "Ingest all of them? [Y/n] (default: Yes): " confirmation_reply
+  confirmation_reply="${confirmation_reply:-Y}"
+
+  if [[ ! "$confirmation_reply" =~ ^[Yy]$ ]]; then
+    echo -e "\nExclude logs by listing their index numbers (e.g. 1, 2, 3):"
+    read exclude_index_list_string
+
+    IFS=', ' read -r -a exclude_index_list_array <<< "$exclude_index_list_string"
+
+    for index in "${!options[@]}"; do
+      local is_excluded=0
+      for excluded_index in "${exclude_index_list_array[@]}"; do
+        if [[ "$index" -eq "$excluded_index" ]]; then
+            is_excluded=1
```

Comment:
I think it would be good to print the log files that were excluded after the exclusion list has been provided for instant feedback. 

It's quite difficult to check at the end whether they got excluded correctly.

So for example:

```text
Exclude logs by listing their index numbers (e.g. 1, 2, 3):
4,6

These logs will not be ingested:

4) /Users/thom/Library/Caches/com.spotify.client/Default/Extension Scripts/*.log
6) /Users/thom/Library/Caches/com.spotify.client/Default/Extension State/*.log
```

Reply:
Good idea, will add this 👍